internal/services/netapp/netapp_backup_policy_test.go

```diff
@@ -0,0 +1,163 @@
+	id, err := backuppolicy.ParseBackupPolicyID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.NetApp.BackupPolicyClient.BackupPoliciesGet(ctx, *id)
+
+	if err != nil {
+		if resp.HttpResponse.StatusCode == http.StatusNotFound {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func TestAccNetAppBackupPolicy_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_netapp_backup_policy", "test")
+	r := NetAppBackupPolicyResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccNetAppBackupPolicy_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_netapp_backup_policy", "test")
+	r := NetAppBackupPolicyResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("2"),
+				check.That(data.ResourceName).Key("enabled").HasValue("true"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("2"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("10"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("10"),
+				check.That(data.ResourceName).Key("daily_backups_to_keep").HasValue("10"),
+				check.That(data.ResourceName).Key("enabled").HasValue("false"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r NetAppBackupPolicyResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_netapp_backup_policy" "test" {
+  name                    = "acctest-NetAppBackupPolicy-%[2]d"
+  resource_group_name     = azurerm_resource_group.test.name
+  location                = azurerm_resource_group.test.location
+  account_name            = azurerm_netapp_account.test.name
+  daily_backups_to_keep   = 2
+  weekly_backups_to_keep  = 2
+  monthly_backups_to_keep = 2
+  enabled                 = true
+
+  tags = {
+    "testTag" = "testTagValue"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
```

Comment:
The `basic` test configuration should only contain properties that are `Required`, this looks more like a `complete` test config to me?

Reply:
good catch, I'll make this complete and add a basic.